### PR TITLE
[BugFix] Lake manual compaction needs to support random partitions

### LINKED ISSUE #41720: Lake manual compaction failed to trigger jobs for all partitions on a table which has random generated partitions
### Steps to reproduce the behavior (Required)
```sql
create table t (id int) properties ("bucket_size"="16");

insert into t values (1);
insert into t values (2);
insert into t values (3);

show partitions from t;
+-------------+---------------+----------------+----------------+-------------+--------+--------------+-------+-----------------+---------+----------+----------+-----------------+------------+-------+-------+-------+
| PartitionId | PartitionName | CompactVersion | VisibleVersion | NextVersion | State  | PartitionKey | Range | DistributionKey | Buckets | DataSize | RowCount | EnableDataCache | AsyncWrite | AvgCS | P50CS | MaxCS |
+-------------+---------------+----------------+----------------+-------------+--------+--------------+-------+-----------------+---------+----------+----------+-----------------+------------+-------+-------+-------+
| 13014       | t             | 0              | 2              | 3           | NORMAL |              |       | ALL KEY         | 1       | 232B     | 1        | true            | false      | 0.00  | 0.00  | 0.00  |
| 13019       | t             | 0              | 2              | 3           | NORMAL |              |       | ALL KEY         | 1       | 232B     | 1        | true            | false      | 0.00  | 0.00  | 0.00  |
| 13006       | t             | 3              | 3              | 4           | NORMAL |              |       | ALL KEY         | 1       | 231B     | 1        | true            | false      | 0.00  | 0.00  | 0.00  |
+-------------+---------------+----------------+----------------+-------------+--------+--------------+-------+-----------------+---------+----------+----------+-----------------+------------+-------+-------+-------+
3 rows in set (0.006 sec)
```

### Expected behavior (Required)

Triggering manual compaction on partition `t` should add 3 compactions jobs.

```sql
show proc '/compactions';
+------------------------------------------+-------+---------------------+---------------------+---------------------+-------+
| Partition                                | TxnID | StartTime           | CommitTime          | FinishTime          | Error |
+------------------------------------------+-------+---------------------+---------------------+---------------------+-------+
| sr_tpcds_bin_partitioned_orc_100.t.13014 | 5007  | 2024-02-27 12:49:53 | 2024-02-27 12:49:54 | 2024-02-27 12:49:54 | NULL  |
| sr_tpcds_bin_partitioned_orc_100.t.13006 | 5008  | 2024-02-27 12:49:53 | 2024-02-27 12:49:54 | 2024-02-27 12:49:54 | NULL  |
| sr_tpcds_bin_partitioned_orc_100.t.13019 | 5009  | 2024-02-27 12:49:53 | 2024-02-27 12:49:54 | 2024-02-27 12:49:54 | NULL  |
+------------------------------------------+-------+---------------------+---------------------+---------------------+-------+
3 rows in set (0.004 sec)
```

### Real behavior (Required)

Triggering manual compaction on partition `t` only add 1 compactions jobs.

```sql
show proc '/compactions';
+------------------------------------------+-------+---------------------+---------------------+---------------------+-------+
| Partition                                | TxnID | StartTime           | CommitTime          | FinishTime          | Error |
+------------------------------------------+-------+---------------------+---------------------+---------------------+-------+
| sr_tpcds_bin_partitioned_orc_100.t.13019 | 5009  | 2024-02-27 12:49:53 | 2024-02-27 12:49:54 | 2024-02-27 12:49:54 | NULL  |
+------------------------------------------+-------+---------------------+---------------------+---------------------+-------+
1 rows in set (0.004 sec)
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - main branch

### PR DESCRIPTION
**Pull request by wxl24life**
## What I'm doing:

Fixes #41720

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [x] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by sduzh — DISMISSED** [review]

**Comment by sduzh** [comment]
@starrocks-xupeng @kevincai PTAL

**Comment by wxl24life** [comment]
@meegoo PLTR

File: fe/fe-core/src/main/java/com/starrocks/alter/CompactionHandler.java [hunk 1]
```diff
@@ -79,10 +79,12 @@ public synchronized ShowResultSet process(List<AlterClause> alterClauses, Databa
         if (RunMode.isSharedDataMode()) {
             List<Partition> allPartitions = findAllPartitions(olapTable, compactionClause);
             for (Partition partition : allPartitions) {
```
Comment (starrocks-xupeng) [inline fe/fe-core/src/main/java/com/starrocks/alter/CompactionHandler.java:81]: need db lock here

Comment (wxl24life) [reply fe/fe-core/src/main/java/com/starrocks/alter/CompactionHandler.java]: > need db lock here

done

**Review by starrocks-xupeng — APPROVED** [review]

**Review by wyb — DISMISSED** [review]

File: fe/fe-core/src/test/java/com/starrocks/alter/CompactionHandlerTest.java [hunk 2]
```diff
@@ -79,68 +73,44 @@ public static void beforeClass() throws Exception {
...
-        }
+        // compaction should trigger on all partitions
+        processAlterClauses(Collections.emptyList(), 2);
     }
 
     @Test(expected = IllegalStateException.class)
     public void testProcessNonCompactionClause() {
         AlterClause nonCompactionClause = mock(AlterClause.class);
-        List<AlterClause> alterList = Collections.singletonList(nonCompactionClause);
+        List<AlterClause> alterList = List.of(nonCompactionClause);
```
Comment (kevincai) [inline fe/fe-core/src/test/java/com/starrocks/alter/CompactionHandlerTest.java:113]: this is ok for jdk9+, will fail compile on jdk8 (for branch-3.2 and early versions)

Comment (wxl24life) [reply fe/fe-core/src/test/java/com/starrocks/alter/CompactionHandlerTest.java]: > this is ok for jdk9+, will fail compile on jdk8 (for branch-3.2 and early versions)

thanks. use collections api instead

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**